### PR TITLE
docs: add topic guide for commands and shells [DET-4901]

### PR DESCRIPTION
**Pull request by dzhu**
## Description

As requested by a community user.

## Test Plan

- [x] render and examine docs

### CODE REVIEW

File: docs/topic-guides/commands-and-shells.txt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+More complex commands including shell constructs can be run as well, as
+long as they are quoted to prevent interpretation by the local shell:
+
+.. code::
+
+   det cmd run 'for x in a b c; do echo $x; done'
+
+``det cmd run`` will stream output from the command until it finishes,
+but the command will continue executing and occupying cluster resources
+even if ``det cmd run`` is interrupted or killed (e.g., due to Control-C
```
Comment (neilconway) [inline docs/topic-guides/commands-and-shells.txt:45]: "``det cmd run``" -> "the CLI" ?

File: docs/topic-guides/commands-and-shells.txt [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+useful settings include:
+
+-  ``bind_mounts``: Specifies directories to be bind-mounted into the
+   container from the host machine. (Due to the structured values
+   required for this setting, it needs to be specified in a config
+   file.)
+
+-  ``resources.slots``: Specifies the number of slots the container will
+   have access to. (Distributed commands and shells are not supported;
+   all slots will be on one machine.)
```
Comment (neilconway) [inline docs/topic-guides/commands-and-shells.txt:129]: Say that if you request more slots than are available on a single machine, the command will never be scheduled?

Mention resource pools?

**Review by neilconway — APPROVED** [review]
This is awesome! Very well written, and thank you for the quick turnaround.

Can we find more places in the docs to add links here? e.g., we could add an FAQ entry?

File: docs/topic-guides/commands-and-shells.txt [hunk 3]
```diff
@@ -0,0 +1,135 @@
+.. _commands-and-shells:
+
+#####################
+ Commands and Shells
+#####################
+
+In addition to structured model training workloads, which are handled
+using :ref:`experiments <experiment-lifecycle>`, Determined also
+supports more free-form tasks using *commands* and *shells*. Commands
+and shells enable users to use a Determined cluster and its GPUs without
```
[the remark this answers is not in the document]
Comment (dzhu) [reply docs/topic-guides/commands-and-shells.txt:10]: I'll go with "developers".

File: docs/topic-guides/commands-and-shells.txt (hunk 1 above)
Comment (dzhu) [reply docs/topic-guides/commands-and-shells.txt:45]: Sounds good!

File: docs/topic-guides/commands-and-shells.txt [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+which runs a command in the cluster and streams its output. For example,
+the following CLI command uses ``nvidia-smi`` to display information
+about the GPUs available to tasks in the container:
+
+.. code::
+
+   det cmd run nvidia-smi
+
+More complex commands including shell constructs can be run as well, as
+long as they are quoted to prevent interpretation by the local shell:
```
[the remark this answers is not in the document]
Comment (dzhu) [reply docs/topic-guides/commands-and-shells.txt:37]: I'd say "interpolation" (i.e., expanding string values of variables) is one specific part of "interpretation" (i.e., all of the syntax and stuff).

File: docs/topic-guides/commands-and-shells.txt (hunk 2 above)
Comment (dzhu) [reply docs/topic-guides/commands-and-shells.txt:129]: > Say that if you request more slots than are available on a single machine, the command will never be scheduled?

Done!

> Mention resource pools?

I think I'll let this be batched in with the other doc updates for resource pools.

**Review by dzhu — COMMENTED** [review]
Couldn't find a spot I liked in the FAQ, but I added links from the terminology page and quick start guide.

**Pull request merged**